core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -154,12 +154,18 @@ protected ShardValidateQueryResponse shardOperation(ShardValidateQueryRequest re
         String explanation = null;
         String error = null;
         ShardSearchLocalRequest shardSearchLocalRequest = new ShardSearchLocalRequest(request.shardId(), request.types(),
-            request.nowInMillis(), request.filteringAliases());
+            request.nowInMillis(), request.filteringAliases(), request.strictFieldNames());
         SearchContext searchContext = searchService.createSearchContext(shardSearchLocalRequest, SearchService.NO_TIMEOUT, null);
         try {
             ParsedQuery parsedQuery = searchContext.getQueryShardContext().toQuery(request.query());
             searchContext.parsedQuery(parsedQuery);
             searchContext.preProcess(request.rewrite());
+            if (request.strictFieldNames()) {
+                Set<String> unmappedFields = searchContext.getQueryShardContext().getUnmappedFields();
+                if (unmappedFields.size() > 0) {
+                    throw new ParsingException(null, "The following fields were unmapped: " + unmappedFields);
```

Comment:
I don't think we know at this stage that all shards failed because it's not a reduce-level operation like the test we have in the search reduction phase?